privilege/privileges/cache.go

```diff
@@ -1019,7 +1019,7 @@ func (p *MySQLPrivilege) showGrants(user, host string, roles []*auth.RoleIdentit
 	}
 	var g string
 	for _, record := range p.User {
-		if record.baseRecord.match(user, host) {
+		if user == record.User && host == record.Host {
```

Comment:
How about creating a new function for this match, for example `record.baseRecord.completeMatch()`?